MusicTaste/MusicTaste/APIManagers/SpotifyAPIManager.m

```diff
@@ -6,9 +6,166 @@
 //
 
 #import "SpotifyAPIManager.h"
+#import <Parse/Parse.h>
+#import "AppDelegate.h"
+#import "MatchmakingViewController.h"
+#import <CoreData/CoreData.h>
+#import "ConnectView.h"
+#import "LoginViewController.h"
 
 @implementation SpotifyAPIManager
 
 
++ (instancetype)shared {
+    static SpotifyAPIManager *sharedManager = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        sharedManager = [[self alloc] init];
+    });
+    return sharedManager;
+}
+
+static NSString * const SpotifyClientID = @"9f9a8a428178497e8c58840c65d9f3c0";
+static NSString * const SpotifySecretID = @"085725985e87480fab42de4970086a54";
+static NSString * const SpotifyRedirectURLString = @"spotify-ios-quick-start://spotify-login-callback";
+
+
+#pragma mark - Actions
+
+- (void) setUpSpotifyWithCompletion:(void (^)(NSDictionary *, NSError*))completion{
+    NSString *spotifyClientID = SpotifyClientID;
+    NSURL *spotifyRedirectURL = [NSURL URLWithString:SpotifyRedirectURLString];
+
+    self.configuration  = [[SPTConfiguration alloc] initWithClientID:spotifyClientID redirectURL:spotifyRedirectURL];
+    
+    self.configuration.playURI = @"";
+
+    self.sessionManager = [[SPTSessionManager alloc] initWithConfiguration:self.configuration delegate:self];
+    
+    SPTScope requestedScope = SPTUserLibraryReadScope | SPTPlaylistReadPrivateScope |  SPTUserTopReadScope;
+     
+    if (@available(iOS 11, *)) {
+        [self.sessionManager initiateSessionWithScope:requestedScope options:SPTDefaultAuthorizationOption];
+    } else {
+        [self.sessionManager initiateSessionWithScope:requestedScope options:SPTDefaultAuthorizationOption presentingViewController:self];
+    }
+    completion(nil, nil);
+}
+
+- (BOOL)application:(UIApplication *)app openURL:(NSURL *)url options:(NSDictionary<UIApplicationOpenURLOptionsKey,id> *)options
+{
+    [self.sessionManager application:app openURL:url options:options];
+    return true;
+}
+
+
+
+#pragma mark - SPTSessionManagerDelegate
+
+- (void)sessionManager:(SPTSessionManager *)manager didInitiateSession:(SPTSession *)session
+{
+    self.token = session.accessToken;
+    [self getSpotifyTracksArtists:^(NSDictionary *dict, NSError *error) {
+    }];
+}
+
+- (void)sessionManager:(SPTSessionManager *)manager didFailWithError:(NSError *)error
+{}
+
+- (void)sessionManager:(SPTSessionManager *)manager didRenewSession:(SPTSession *)session
+{}
+
+
+
+
+
+#pragma mark - Getting Spotify Data
+
+-(void) getSpotifyTracksArtists:(void (^)(NSDictionary *, NSError*))completion{
+    // get artists
+    [self getSpotifyData:@"https://api.spotify.com/v1/me/top/artists" completion:^(NSDictionary * artistDict, NSError * error) {
+        if (!error){
```

Comment:
What if there is an error; would you like to log that maybe? Or display a UI alert?